.github/workflows/Tests.yaml

```diff
@@ -433,3 +434,129 @@ jobs:
           # We get occasional random timeouts, repeat tests to see if
           # it is a random timeout or systematic
           ctest -j2 --output-on-failure -L inputfiles --repeat after-timeout:3
+
+  arch_datastructures_tests:
+    name: DataStructures arch tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [gcc-9,
+                   clang-10
```

Comment:
For my understanding, how relevant is the choice of compiler for this whole archs-issue?

Reply:
It is at least somewhat important, because different compilers will bring in different vectorization flags, and that can influence which compile-time branches Blaze uses in the implementation.
I've chosen a couple of the most modern ones under the assumption that these are the most rapidly changing ones, and most prone to make unexpected alterations in how Blaze performs, but I'm happy to make adjustments for which compilers we test on.

Reply:
Do we have specific cluster/compiler/arch combinations in production that would be important to test, e.g. the case that prompted the revert?